device-connectors/src/testflinger_device_connectors/devices/zapper_kvm/__init__.py

```diff
@@ -63,25 +64,57 @@ def _get_autoinstall_conf(self) -> Optional[Dict[str, Any]]:
         SSH public key.
         """
         autoinstall_conf = {}
-        for key, value in self.job_data["provision_data"].items():
-            if "autoinstall_" not in key:
-                continue
 
-            autoinstall_key = key.replace("autoinstall_", "")
-            with contextlib.suppress(AttributeError):
-                getattr(self, f"_validate_{autoinstall_key}")(value)
+        if self.job_data["provision_data"].get("autoinstall_oem"):
+            # In case of Ubuntu OEM via autoinstall, use a
+            # dedicated user-data file as base.
 
-            autoinstall_conf[autoinstall_key] = value
+            logger.info(
+                "When using 'autoinstall_oem', "
+                "other autoinstall keys are not considered"
+            )
+
+            oem_autoinstall_data = (
+                Path(__file__).parent / "../../data/zapper_kvm"
+            )
+            with open(
+                oem_autoinstall_data / "user-data-oem", "r"
+            ) as default_user_data:
+                user_data = default_user_data.read()
+                encoded_user_data = base64.b64encode(user_data.encode())
+
+            user_data = yaml.safe_load(user_data)
+            storage = user_data["autoinstall"]["storage"]["layout"]["name"]
+
+            autoinstall_conf = {
+                "base_user_data": encoded_user_data,
+                "storage_layout": storage,
+            }
+
+        else:
+            for key, value in self.job_data["provision_data"].items():
+                if "autoinstall_" not in key:
+                    continue
+
+                autoinstall_key = key.replace("autoinstall_", "")
+
+                with contextlib.suppress(AttributeError):
+                    getattr(self, f"_validate_{autoinstall_key}")(value)
+
+                autoinstall_conf[autoinstall_key] = value
```

Comment:
This looks like a single logical block. Additionally, I see the `getattr` line is doing nothing?

```suggestion
                autoinstall_key = key.replace("autoinstall_", "")
                with contextlib.suppress(AttributeError):
                    getattr(self, f"_validate_{autoinstall_key}")(value)
                autoinstall_conf[autoinstall_key] = value
```